demo/site/src/blocks/CookieSafeYouTubeVideoBlock.tsx

```diff
@@ -0,0 +1,25 @@
+import { CookieSafe, PropsWithData, YouTubeVideoBlock } from "@comet/cms-site";
+import { YouTubeVideoBlockData } from "@src/blocks.generated";
+import { CookieFallback } from "@src/components/common/CookieFallback";
+import { cookieIds, useCookieApi } from "@src/util/cookies";
+import styled from "styled-components";
+
+// TODO: Consider if `CookieSafe` should be used directly inside `YouTubeVideoBlock` and expose the `consented` and `fallback` props.
```

Comment:
We just moved the YouTubeVideoBlock to the library, I'm not sure if we should require a custom block in the application

Reply:
So you think we should use `<CookieSafe />` directly inside the existing `YouTubeVideoBlock` and expose the necessary props?

Reply:
It depends whether we want our developers to always make sure that cookies are accepted. Then we should add it to the YouTubeVideo block.

Reply:
> It depends whether we want our developers to always make sure that cookies are accepted.

I think: yes

Reply:
yes, the library blocks should have support for CookieSafe out of the box, with no additional configuration needed in the application.

You will have to come up with a different solution for defining the used cookie api (possibly using a context) in order to be able to implement that.

Reply:
Then I'll change it back to use a context 😄 
Of course, you still would need to define which API to use in the `_app.tsx`, where the provider is used.

Reply:
> yes, the library blocks should have support for CookieSafe out of the box, with no additional configuration needed in the application.

Should we add support to the comet blocks (e.g. `YouTubeVideoBlock`) in a separate PR, as those would be breaking changes?

Reply:
> Should we add support to the comet blocks (e.g. YouTubeVideoBlock) in a separate PR, as those would be breaking changes?

Probably.

Reply:
I created a task for this: AG-378